app/Mage.php

```diff
@@ -751,6 +751,7 @@ public static function run($code = '', $type = 'store', $options = [])
             die();
         } catch (Exception $e) {
             if (self::isInstalled()) {
+                self::dispatchEvent('mage_installed_exception', ['exception' => $e]);
```

Comment:
@kiatng do we have anything against this? would a different name be better?

Reply:
![image](https://github.com/OpenMage/magento-lts/assets/1106470/7a5bdef3-3876-49e1-97b1-b5c2525e97c4)
If the first `if` block is removed, then there is already an event, which will be dispatched installed or not.

Reply:
I think the upper if (773 to 775) acts as a guard and is okay to stay.
Also because we immediately exit the program and not change "Location" to somewhere possibly outside of (current) context/control.

Reply:
I tested the following:

```php
           // remove the first if block lines 773-776
            try {
                self::dispatchEvent('mage_run_exception', ['exception' => $e]);
                if (!headers_sent() && !self::isInstalled()) {
               // ...
```

You can do a test easily with `throw new Exception('test');` in any controller action. Not sure how to test installation but the check on `!isInstalled()` seems correct, without the not, it would redirect wrongly.

Reply:
I'm not sure about @kiatng, if it requires a modification of if it's ok as it is.

also, @pquerner do you think we could find a better name for the event? to me mage_installed_exception doesn't feel very clear, what about mage_run_exception_installed or mage_run_installed_exception, i'm not sure even about these ones...

Reply:
We have merged this in our core as event `mage_installed_run_exception` by now. I didnt touch this PR here and update yet again its name, because I sensed it would come up in a future discussion. ^^

Like I said, I like the guards in place as they currently are, but I can see why @kiatng wants it changed. I'd be fine with either.

Reply:
i'd kinda prefer mage_run_installed_exception cause keeps "mage_run" as "class_method" prefix, not that that's a standard.

in case of @kiatng's code, this new event wouldn't have any place to go right?

dunno... i guess it's a matter of tossing a coin?

Reply:
Would you expect a "printing" method to throw new events? To me thats a no. Although I could live with it. But its not really expected.

Reply:
mmm that's an interesting point

Reply:
```suggestion
                self::dispatchEvent('mage_run_installed_exception', ['exception' => $e]);
```